packages/flutter_tools/test/general.shard/mdns_discovery_test.dart

```diff
@@ -254,17 +230,74 @@ void main() {
         },
       );
 
-      final MockIOSDevice mockDevice = MockIOSDevice();
-      when(mockDevice.portForwarder).thenReturn(const NoOpDevicePortForwarder());
+      final FakeIOSDevice device = FakeIOSDevice();
       final MDnsObservatoryDiscovery portDiscovery = MDnsObservatoryDiscovery(
         mdnsClient: client,
         logger: BufferLogger.test(),
         flutterUsage: TestUsage(),
       );
-      final Uri uri = await portDiscovery.getObservatoryUri('bar', mockDevice, hostVmservicePort: 0);
+      final Uri uri = await portDiscovery.getObservatoryUri('bar', device, hostVmservicePort: 0);
       expect(uri.toString(), 'http://127.0.0.1:123/');
     });
   });
 }
 
-class MockMDnsClient extends Mock implements MDnsClient {}
+class FakeMDnsClient extends Fake implements MDnsClient {
```

Comment:
When these `Fake`s get elaborate I'm always worried we're testing the fake class and not the actual interactions in prod.  If only we had a mocking framework...  But it's the same concern for complicated mocks.

Reply:
Its important to remember that unit tests really just exist to catch the most trivial of errors

Reply:
I'm just pontificating